### PR TITLE
fix(k8s): fix waiting for multicloud cluster

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: scaleway/resource_k8s_cluster.go [hunk 1]
```diff
@@ -385,8 +385,13 @@ func resourceScalewayK8SClusterCreate(ctx context.Context, d *schema.ResourceDat
 	}
 
 	d.SetId(newRegionalIDString(region, res.ID))
-
-	_, err = waitK8SClusterPool(ctx, k8sAPI, region, res.ID, d.Timeout(schema.TimeoutCreate))
+	if clusterType.(string) == "multicloud" {
+		// In case of multi-cloud, we do not have the guarantee that a pool will be created in Scaleway.
+		_, err = waitK8SCluster(ctx, k8sAPI, region, res.ID, d.Timeout(schema.TimeoutCreate))
```
Comment (Codelax) [inline scaleway/resource_k8s_cluster.go:390]: `waitK8SCluster` do the same thing as `waitK8SClusterPool` but allows more terminal status, why not using `waitK8SCluster` for standard and multicloud cluster

Comment (Codelax) [inline scaleway/resource_k8s_cluster.go:390]: `waitK8SCluster` do the same thing as `waitK8SClusterPool` but allows more terminal status, why not using `waitK8SCluster` for standard and multicloud cluster

Comment (Monitob) [reply scaleway/resource_k8s_cluster.go:390]: `waitK8SClusterPool`  will wait that the pool is created but when we threat `multicloud` the pools can be elsewhere.

File: scaleway/resource_k8s_cluster.go [hunk 2]
```diff
@@ -385,8 +385,13 @@ func resourceScalewayK8SClusterCreate(ctx context.Context, d *schema.ResourceDat
 	}
 
 	d.SetId(newRegionalIDString(region, res.ID))
-
-	_, err = waitK8SClusterPool(ctx, k8sAPI, region, res.ID, d.Timeout(schema.TimeoutCreate))
+	if clusterType.(string) == "multicloud" {
```
Comment (Monitob) [inline scaleway/resource_k8s_cluster.go:388]: Can we set the types on the SDK? or as const values

File: scaleway/resource_k8s_cluster.go (hunk 1 above)
Comment (Monitob) [reply scaleway/resource_k8s_cluster.go:390]: How can we handle the transient states on multicloud?

Comment (Codelax) [reply scaleway/resource_k8s_cluster.go:390]:
```go
func (s *API) WaitForClusterPool(req *WaitForClusterRequest, opts ...scw.RequestOption) (*Cluster, error) {
	cluster, err := async.WaitSync(&async.WaitSyncConfig{
		Get: func() (interface{}, bool, error) {
			cluster, err := s.GetCluster(&GetClusterRequest{
				ClusterID: req.ClusterID,
				Region:    req.Region,
			}, opts...)
			if err != nil {
				return nil, false, err
			}

			_, isTerminal := terminalStatus[cluster.Status]
			return cluster, isTerminal, nil
		},
		Timeout:          timeout,
		IntervalStrategy: async.LinearIntervalStrategy(retryInterval),
	})
}
```

`waitK8SClusterPool` will not wait for pool, it is the same as `waitK8SCluster` but with different terminal state

For `waitK8SClusterPool`
```go
	terminalStatus := map[ClusterStatus]struct{}{
		ClusterStatusPoolRequired: {},
	}
```

For `waitK8SCluster`
```go
	terminalStatus := map[ClusterStatus]struct{}{
		ClusterStatusReady:        {},
		ClusterStatusLocked:       {},
		ClusterStatusDeleted:      {},
		ClusterStatusPoolRequired: {},
	}
```

Comment (Monitob) [reply scaleway/resource_k8s_cluster.go:390]: Ok for the creation was necessary only check the pool creation. It's a special case. I will not be the case for multicloud

File: scaleway/resource_k8s_cluster.go (hunk 2 above)
Comment (remyleone) [reply scaleway/resource_k8s_cluster.go:388]: Nope they are like DEV1-S. We don't enum them because they can change more often.

**Review by Monitob — APPROVED** [review]

**Pull request merged**